src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs

```diff
@@ -176,7 +176,37 @@ public IBindingProvider BindToGenericItem<TAttribute>(Func<TAttribute, Type, Tas
             Func<TAttribute, ParameterInfo, INameResolver, Task<TAttribute>> postResolveHook = null)
             where TAttribute : Attribute
         {
-            var bindingProvider = new ExactTypeBindingProvider<TAttribute, TUserType>(_nameResolver, buildFromAttribute, buildParameterDescriptor, postResolveHook);
+            var bindingProvider = new ExactTypeBindingProvider<TAttribute, TUserType>(
+                _nameResolver, 
+                buildFromAttribute,
+                null,
+                buildParameterDescriptor, 
+                postResolveHook);
+            return bindingProvider;
+        }
+
+        /// <summary>
+        /// Create a binding provider that binds the parameter to a set of possible types via converter manager. 
+        /// </summary>
+        /// <typeparam name="TAttribute">type of binding attribute on the user's parameter.</typeparam>
+        /// <typeparam name="TMiddleType">An intermediate type to bind to. 
+        /// Converter Manager will then convert this intermediate type to the user's exact type in their signature.</typeparam>
+        /// <param name="buildFromAttribute">builder function to create the object that will get passed to the user function.</param>
+        /// <param name="buildParameterDescriptor">An optional function to create a specific ParameterDescriptor object for the dashboard. If missing, a default ParameterDescriptor is created. </param>
+        /// <param name="postResolveHook">an advanced hook for translating the attribute. </param>
+        /// <returns>A binding provider that applies these semantics.</returns>
+        public IBindingProvider BindToGeneralAsyncType<TAttribute, TMiddleType>(
```

Comment:
For many other methods on this type you use the terminology "BindToGenericxxx" - can you use that here as well? Otherwise the differences between BindToGeneric and BindToGeneral aren't clear